core/src/main/java/com/oath/oak/Chunk.java

```diff
@@ -459,7 +461,7 @@ int linkEntry(int ei, boolean cas, K key) {
                       }
                     }
                   }
-                  return ei;
+                  return new LinkEntryResult(ei, true);
                 }
                 // CAS didn't succeed, try again
             } else {
```

Comment:
why not return in else?

Reply:
Because we do the loop again and try to find the entry that did get to the cas first